src/plugins/cordova-plugin-geolocation/sim-host.js

```diff
@@ -77,6 +88,66 @@ module.exports = function(messages) {
         }
     }
 
+    function mapEventTelemetryHandler() {
+        pendingMouseEvents++;
+        setTimeout(function () {
+            --pendingMouseEvents;
+
+            if (pendingMouseEvents === 0) {
+                telemetry.sendUITelemetry(Object.assign({}, baseProps, { control: 'geo-map-container' }));
+            }
+        }, mouseEventHoldDelay);
+    }
+
+    function registerTelemetryEvents() {
+        // Register the simple events (onclick / onchange / etc -> send the control ID).
+        var basicTelemetryEvents = [
+            { control: 'geo-latitude' },
+            { control: 'geo-longitude' },
+            { control: 'geo-altitude' },
+            { control: 'geo-accuracy' },
+            { control: 'geo-altitude-accuracy' },
+            { control: 'geo-heading' },
+            { control: 'geo-speed' },
+            { control: 'geo-delay' },
+            { control: 'geo-gpxfile-button', event: 'click' },
+            { control: 'geo-map-zoom-decrease', event: 'click' },
+            { control: 'geo-map-zoom-increase', event: 'click' }
+        ];
+
+        basicTelemetryEvents.forEach(function (controlEvent) {
+            registerTelemetryForControl(controlEvent.control, controlEvent.event);
+        });
+
+        // Register the event for the timeout checkbox.
+        // Clicking the checkbox's label fires the click event twice, so keep track of the previous state. Note that we can't use the change event because the component seems to swallow it.
+        var previousTimeoutState = false;
+        var geoTimeoutCheckbox = document.querySelector('#geo-timeout');
+
+        geoTimeoutCheckbox.onclick = function () {
+            if (geoTimeoutCheckbox.checked !== previousTimeoutState) {
+                previousTimeoutState = geoTimeoutCheckbox.checked;
+                telemetry.sendUITelemetry(Object.assign({}, baseProps, { control: 'geo-timeout' }));
+            }
+        };
+
+        // Register the event for the Go button.
+        document.getElementById('geo-gpx-go').onclick = function () {
+            var rateList = document.getElementById('geo-gpxmultiplier-select');
+            var option = rateList.options[rateList.selectedIndex];
+
+            telemetry.sendUITelemetry(Object.assign({}, baseProps, { control: 'geo-gpx-go', value: option.value }));
+        }
+
+        // Register the event for zooming with the mouse wheel on the map.
+        document.getElementById('geo-map-container').onwheel = mapEventTelemetryHandler;
+    }
+
+    function registerTelemetryForControl(controlId, event) {
+        event = event || 'change';
+        document.getElementById(controlId).addEventListener(event, telemetry.sendUITelemetry.bind(this, Object.assign({}, baseProps, { control: controlId })));
+    }
+
```

Comment:
I agree that it would be ideal, but in practice I find it hard to implement. I tried keeping infrastructure-related code in its own files and modules, but for the actual telemetry points / events, it's hard to be decoupled. We need direct access to the DOM elements so that we can hook up event handlers to some of their events, and giving a separate module access to the DOM might be even more complicated to maintain.

I am fully open to design suggestions that might accomplish what you are describing, but for now I don't know how to easily improve the current approach (having the UI code register its own telemetry events).